CMakeLists.txt

```diff
@@ -96,11 +98,20 @@ endfunction()
 # x86-64
 if (ARCH_ID STREQUAL "x86_64" OR ARCH_ID STREQUAL "x86-64" OR ARCH_ID STREQUAL "amd64")
   list(APPEND randomx_sources
-    src/jit_compiler_x86_static.S
     src/jit_compiler_x86.cpp)
-  # cheat because cmake and ccache hate each other
-  set_property(SOURCE src/jit_compiler_x86_static.S PROPERTY LANGUAGE C)
-  set_property(SOURCE src/jit_compiler_x86_static.S PROPERTY XCODE_EXPLICIT_FILE_TYPE sourcecode.asm)
+
+  if (MSVC)
+    enable_language(ASM_MASM)
+    list(APPEND randomx_sources src/jit_compiler_x86_static.asm)
+
+    set_property(SOURCE src/jit_compiler_x86_static.asm PROPERTY LANGUAGE ASM_MASM)
+  else()
+    list(APPEND randomx_sources src/jit_compiler_x86_static.S)
+
+    # cheat because cmake and ccache hate each other
+    set_property(SOURCE src/jit_compiler_x86_static.S PROPERTY LANGUAGE C)
+    set_property(SOURCE src/jit_compiler_x86_static.S PROPERTY XCODE_EXPLICIT_FILE_TYPE sourcecode.asm)
+  endif()
```

Comment:
You need to also explicitly enable AVX2 instruction set for src/argon2_avx2.c, otherwise AVX2 will not work.

Reply:
Proper check for availability of AVX2 instructions for MSVC is not so trivial, because requires runtime check (check this gist https://gist.github.com/UnaNancyOwen/263c243ae1e05a2f9d0e).
I think this is out of scope for this change and should be done in a separate change.

Reply:
RandomX already does a [runtime check](https://github.com/tevador/RandomX/blob/master/src/randomx.cpp#L53). It's enough to unconditionally compile src/argon2_avx2.c with AVX2 instructions turned on, just like [we do with GCC](https://github.com/tevador/RandomX/blob/master/CMakeLists.txt#L116).

Be careful to not enable AVX2 for the whole build. It's safe only for this one file (src/argon2_avx2.c).

Reply:
Done